internal/backend/remote-state/s3/client.go

```diff
@@ -295,6 +304,72 @@ func (c *RemoteClient) Lock(info *statemgr.LockInfo) (string, error) {
 
 	log.Info("Locking remote state")
 
+	var fileLockID, dynamoDBLockID string
+	if c.useLockFile {
+		fileLockID, err = c.lockWithFile(ctx, info, log)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if c.ddbTable != "" {
+		dynamoDBLockID, err = c.lockWithDynamoDB(ctx, info, log)
+		if err != nil {
+			// Release the file lock if attempting to acquire the DynamoDB lock fails.
+			if c.useLockFile {
+				if unlockErr := c.unlockWithFile(ctx, fileLockID, &statemgr.LockError{}, log); unlockErr != nil {
+					err = fmt.Errorf("error when attempting to clean up the file lock after failing to acquire DynamoDB lock: %v; original DynamoDB lock error: %w", unlockErr, err)
+				}
+			}
+			return "", err
+		}
+	}
+
+	// Combine file and DynamoDB lock IDs using a delimiter to support using both locking mechanisms at the same time.
+	return joinLockIDs(fileLockID, dynamoDBLockID), err
```

Comment:
Since the lock ID is either passed in as `info.ID` or set above, both `fileLockID` and `dynamoDBLockID` will be the same. Since they're the same lock, they should have the same lock ID.

```suggestion
	return info.ID, err
```

Reply:
fixed in [e246ad3](https://github.com/hashicorp/terraform/pull/35661/commits/e246ad32f9711d4433220426bba9ff4b2da5fe92)